### PR TITLE
feat(community): add libsql vectorstore

### PR DESCRIPTION
**Pull request by notrab**
This PR adds libSQL as a vectorstore option with some basic examples and docs.

**Notes**

* It currently doesn't properly support metadata filtering, but we're seeking ways with our community on the best developer experience for this and will follow up on this PR.
* I could not run the npx command to initialize the new integration folder. So I did what I could manually to replicate.

### CODE REVIEW

File: docs/core_docs/docs/integrations/vectorstores/libsql.mdx [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# libSQL
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/vectorstores/libsql.mdx:1]: Can we make the docs follow this template?

https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-scripts/src/cli/docs/templates/vectorstores.ipynb

I need to update the contributing instructions...

File: libs/langchain-community/package.json [hunk 2]
```diff
@@ -82,6 +82,7 @@
     "@langchain/scripts": ">=0.1.0 <0.2.0",
     "@langchain/standard-tests": "0.0.0",
     "@layerup/layerup-security": "^1.5.12",
+    "@libsql/client": "^0.14.0",
```
Comment (jacoblee93) [inline libs/langchain-community/package.json:85]: This also needs to be an optional peer dep:

https://js.langchain.com/docs/contributing/code/#adding-an-entrypoint

File: libs/langchain-community/src/vectorstores/libsql.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  }
+
+  /**
+   * Performs a similarity search using a vector query and returns documents with their scores.
+   * @param {number[]} query - The query vector.
+   * @param {number} k - The number of results to return.
+   * @returns {Promise<[Document, number][]>} An array of tuples containing the similar documents and their scores.
+   */
+  async similaritySearchVectorWithScore(
+    query: number[],
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/libsql.ts:126]: Add a comment warning about potential SQL injection, or manually sanitize the input to make sure everything is a number.

**Comment by jacoblee93** [comment]
Hey @notrab, sorry for the delay - left a few minor comments but overall looks good.

I may be able to spend some time reformatting things next week if you don't have time.

**Comment by notrab** [comment]
@jacoblee93 perfect, thank you. I’ll take another pass at this Monday/Tuesday and see what you think. Thanks very much

File: libs/langchain-community/package.json (hunk 2 above)
Comment (notrab) [reply libs/langchain-community/package.json:85]: Added

File: docs/core_docs/docs/integrations/vectorstores/libsql.mdx (hunk 1 above)
Comment (notrab) [reply docs/core_docs/docs/integrations/vectorstores/libsql.mdx:1]: @jacoblee93 I updated the docs to follow more closely your example, however if you have time, please suggest or commit any changes you desire.

**Comment by jacoblee93** [comment]
Thanks! Will ship tonight.

**Pull request merged**